godot_project/project_workspace/structs/nano_particle_emitter.gd

```diff
@@ -123,33 +180,47 @@ func create_instances(out_group: AtomicStructure) -> void:
 				elements[atom_idx],
 				_transform.origin + positions[atom_idx] + offset
 			)
-			var new_atom_id: int = out_group.add_atom(params)
+			var new_atom_id: int = _instances_group.add_atom(params)
 			instance_atom_map[atom_idx] = new_atom_id
 			this_instance_atoms.push_back(new_atom_id)
 		for bond_idx: int in bonds.size():
 			var atom1: int = instance_atom_map[bonds[bond_idx].x]
 			var atom2: int = instance_atom_map[bonds[bond_idx].y]
 			var bond_order: int = bonds[bond_idx].z
-			var new_bond_id: int = out_group.add_bond(atom1, atom2, bond_order)
+			var new_bond_id: int = _instances_group.add_bond(atom1, atom2, bond_order)
 			this_instance_bonds.push_back(new_bond_id)
 	_instances_group.end_edit()
+	
+	# Mark all atoms as invalid
+	# TODO: this could use a specific API instead of relying on the current implementation details
+	# TODO: currently requires to flush all the previous commands by calling end / start_edit
+	_instances_group.start_edit()
+	for instance: PackedInt32Array in _instances_atom_ids:
+		for atom_id: int in instance:
+			if _instances_group.is_atom_valid(atom_id):
+				_instances_group.remove_atom(atom_id)
+	for instance: PackedInt32Array in _instances_bond_ids:
+		for bond_id: int in instance:
+			if _instances_group.is_bond_valid(bond_id):
+				_instances_group.remove_bond(bond_id)
+	_instances_group.end_edit()
```

Comment:
Making all atoms invalid during the same step as we're creating them doesn't work.
It works when flushing the previous commands and starting to edit again, but maybe it's worth creating a dedicated api for that?

Reply:
remove_atom/bond could check if the atom_id is in the "added" queue and remove it from there instead of adding it to the removed queue